app/src/main/java/org/greenstand/android/TreeTracker/managers/Preferences.kt

```diff
@@ -0,0 +1,99 @@
+package org.greenstand.android.TreeTracker.managers
+
+import android.content.SharedPreferences
+
+object PrefKeys {
+
+    val ROOT = PrefKey("greenstand")
+
+    val SESSION = PrefKey("session")
+
+    val USER_SETTINGS = ROOT + PrefKey("user-settings")
+}
+
+open class PrefKey(val path: String) {
+
+    operator fun plus(prefKey: PrefKey): PrefKey {
+        return when (prefKey) {
+            is UserPrefKey -> UserPrefKey("$path/${prefKey.path}")
+            else -> PrefKey("$path/${prefKey.path}")
+        }
+    }
+
+    fun asUserPref(): UserPrefKey {
+        return UserPrefKey(path)
+    }
+}
+
+class UserPrefKey(path: String) : PrefKey(path)
```

Comment:
I tried this out, but it would require either a method only on the UserPrefKey, or an overridden one that does nothing on the normal key. 
I think it's nice to keep the keys as simple data holders without logic.